gst/gst_element.go

```diff
@@ -340,6 +340,22 @@ func (e *Element) GetState() State {
 	return State(e.Instance().current_state)
 }
 
+// GetStateWithTimeout blocks for the given timeout until the given state is reached.
+//
+// returns true on
+func (e *Element) GetStateWithTimeout(state State, timeout ClockTime) bool {
```

Comment:
Should this be `BlockGetStateWithTimeout`?

What do we do with other functions like this? But I don't like unnecessary deviation from the gstreamer function name...

We haven't specified a 1.0 of this new version yet so is now a good time to name this `GetState` and force passing in a timeout and change the other GetState to be non standard like "GetStateFromInstance" or something?

Reply:
I was hoping you say this ;)

In that case this function should return the "state change return" instead of a boolean though.

Should we also return the pending and current state? I don't fully understand why they are passed out via pointers in the original function.